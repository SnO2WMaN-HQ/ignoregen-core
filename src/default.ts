import {Option} from './types';

export const defaultOption: Option = {
  src:
    'https://raw.githubusercontent.com/SnO2WMaN-HQ/ignoregen-template/master/templates/',
};
