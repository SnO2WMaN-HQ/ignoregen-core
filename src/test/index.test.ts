/* eslint-disable @shopify/jest/no-snapshots */
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import * as fs from 'fs';
import path from 'path';
import url from 'url';
import {promisify} from 'util';
import {parse} from '..';
import {defaultOption} from '../default';

const axiosMock = new MockAdapter(axios);

const readdir = promisify(fs.readdir);
const readFile = promisify(fs.readFile);

const fixturesPath = path.resolve(__dirname, 'fixtures');
const templatesPath = path.resolve(fixturesPath, 'templates');

describe('local fixture test', () => {
  beforeEach(async () => {
    const dirents = await (
      await readdir(templatesPath, {withFileTypes: true})
    ).filter((dirent) => dirent.isFile());

    for (const {name} of dirents) {
      const content = await readFile(path.resolve(templatesPath, name), {
        encoding: 'utf-8',
      });

      axiosMock.onGet(url.resolve(defaultOption.src, name)).reply(200, content);
    }
  });

  afterEach(() => {
    axiosMock.reset();
  });

  it.each([['1.ignore'], ['2.ignore'], ['3.ignore']])(
    'check for %s',
    async (targetFile) => {
      const target = (
        await readFile(path.resolve(fixturesPath, targetFile), 'utf-8')
      ).split('\n');

      expect(await parse(target)).toMatchSnapshot();
    },
  );
});
