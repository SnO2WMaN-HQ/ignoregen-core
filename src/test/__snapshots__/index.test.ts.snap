// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local fixture test check for 1.ignore match snapshot 1`] = `
Array [
  "# ignoregen node",
  "node_modules/",
  "",
  "# ignoregen env",
  ".env*",
  ".envrc",
  "!.env.example",
  "",
]
`;

exports[`local fixture test check for 2.ignore match snapshot 1`] = `
Array [
  "dist",
  "",
]
`;

exports[`local fixture test check for 3.ignore match snapshot 1`] = `
Array [
  "# ignoregen env",
  ".env*",
  ".envrc",
  "!.env.example",
  "",
  "dist/",
  "",
  "# ignoregen node",
  "node_modules/",
  "",
  ".vscode/",
  "",
]
`;
